packages/rest/src/lib/RequestManager.ts

```diff
@@ -425,18 +423,9 @@ export class RequestManager extends EventEmitter {
 				// FormData.append only accepts a string or Blob.
 				// https://developer.mozilla.org/en-US/docs/Web/API/Blob/Blob#parameters
 				// The Blob constructor accepts TypedArray/ArrayBuffer, strings, and Blobs.
-				if (Buffer.isBuffer(file.data)) {
-					// Try to infer the content type from the buffer if one isn't passed
-					const { fileTypeFromBuffer } = await getFileType();
-					let contentType = file.contentType;
-					if (!contentType) {
-						const parsedType = (await fileTypeFromBuffer(file.data))?.mime;
-						if (parsedType) {
-							contentType = OverwrittenMimeTypes[parsedType as keyof typeof OverwrittenMimeTypes] ?? parsedType;
-						}
-					}
 
-					formData.append(fileKey, new Blob([file.data], { type: contentType }), file.name);
+				if (typeof Buffer !== 'undefined' && Buffer.isBuffer(file.data) && isAPNG(file.data)) {
```

Comment:
But what ifff the data is Uint8Array/ArrayBuffer? It'll get stringified, which it shouldn't be. You also completely changed the logic here, the content type should be overriden only for APNG, otherwise... I guess we could try defaulting to just application/octet-stream?

Reply:
Also, isAPNG does the previous checks anyways (its types should probably accept any since it thoroughly checks), so they're happening twice.

And actually lettign isAPNG handle that would solve the problem with Uint8Arrays right?